src/component/RoadmapCanvas.js

```diff
@@ -3,101 +3,69 @@ import ContextMenu from "d3-v4-contextmenu";
 
 export class RoadmapCanvas {
   type;
-
   targetElement;
 
+  style = {
+    barHeight: 20,
+    topPadding: 20
+  };
+
   constructor(options) {
     this.type = options.type;
     this.targetElement = d3.select("#" + options.target);
+    this._setStyle(options.style !== undefined ? options.style : {});
+  }
+
+  _setStyle(style) {
+    if (style.barHeight !== undefined) {
+      this.style.barHeight = parseInt(style.barHeight);
+    }
+    if (style.topPadding !== undefined) {
+      this.style.topPadding = parseInt(style.topPadding);
+    }
+    this.style.gap = this.style.barHeight + 4;
   }
 
   render(roadmap) {
-    if (this.type === "taks") {
-      this._render(roadmap.getTasks(), roadmap);
+    if (this.type === "tasks") {
+      this._render(roadmap.getTasks(), roadmap.getTaskGroups(), roadmap);
     } else if (this.type === "people") {
-      this._render(roadmap.getPeople(), roadmap);
+      this._render(roadmap.getPeople(), roadmap.getPersonGroups(), roadmap);
     } else {
-      this._render(roadmap.getTasks(), roadmap);
-      this._render(roadmap.getPeople(), roadmap);
+      this._render(roadmap.getTasks(), roadmap.getTaskGroups(), roadmap);
+      this._render(roadmap.getPeople(), roadmap.getPersonGroups(), roadmap);
     }
   }
 
-  _render(items, roadmap) {
-    const barHeight = 20;
-    const gap = barHeight + 4;
-    const topPadding = 20;
-
-    const h = items.length * gap + 40;
+  _render(items, groups, roadmap) {
     const w = this.targetElement.node().clientWidth;
+    const h = items.length * this.style.gap + 40;
 
-    let svg = this.targetElement.append("svg").attr("width", w).attr("style", "overflow: visible");
-    svg.attr("height", function() {
-      return parseInt(svg.attr("height") || 0, 10) + h;
-    });
+    let svg = this._generateSVG(w, h);
```

Comment:
i tried to change, but it's too hard...
because i think `d3.js` is not suitable for your suggestion